### DISCUSSION TITLE
Tuning Questions

### DISCUSSION THREAD

**Discussion by baffles** [blakeblackshear/frigate#14511] [Ask A Question] [answered]
I've been working on tuning a new Frigate install, and two questions keep coming up for me. I don't see either answered in the docs, so I'm not sure if I'm just approaching things incorrectly or missing something obvious...

The first is: is there a way to view something like the debug view historically? Being able to replay a section of a recording in the debug view as if it were live would be nice, so that I'd be able to see exactly what happened to trigger an event, a detection change, etc. (and also being able to replay an event to see how it behaves after tweaking the config). For example, in certain lighting conditions, I'm getting extra events from stationary objects coming and going. I'm assuming this is due to changing in lighting being classified as motion, and then the detector runs and for some reason my stationary object goes away. The part I'm struggling with is how to tweak the threshold in a sane way if I can't see what the confidence value at a particular point in time was. It would be nice to be able to replay a recording and see what the detector was doing then; otherwise, I feel like I'm just making up a number and guessing until I get it right. It seems like there's a better way that I'm missing.

The other question I have is if there's a way to make the debug view show everything, so if something _isn't_ being detected, I can see if it is there with a lower confidence or if it's just missing entirely. For example, if I'm seeing an object on one of my cameras that I want to be detected -- I don't know how to tell why it isn't being detected. I.e., do I need to train my Frigate+ model more/do I need to tweak the min score or threshold/is there something else wrong? I can keep adjusting the object parameters to try and get a detection, but again, that feels like a big guessing game and like I'm missing an obvious better approach.

I'm feeling very limited to needing to try and observe things live to view this sort of information, so I feel like I'm missing something obvious here and taking the wrong approach to tuning.

**Comment by hawkeye217** [comment, marked as answer]
> I've been working on tuning a new Frigate install, and two questions keep coming up for me. I don't see either answered in the docs, so I'm not sure if I'm just approaching things incorrectly or missing something obvious...
> 
> The first is: is there a way to view something like the debug view historically? Being able to replay a section of a recording in the debug view as if it were live would be nice, so that I'd be able to see exactly what happened to trigger an event, a detection change, etc. (and also being able to replay an event to see how it behaves after tweaking the config). For example, in certain lighting conditions, I'm getting extra events from stationary objects coming and going. I'm assuming this is due to changing in lighting being classified as motion, and then the detector runs and for some reason my stationary object goes away. The part I'm struggling with is how to tweak the threshold in a sane way if I can't see what the confidence value at a particular point in time was. It would be nice to be able to replay a recording and see what the detector was doing then; otherwise, I feel like I'm just making up a number and guessing until I get it right. It seems like there's a better way that I'm missing.

Yes, this is coming in the next version of Frigate as the "Object Lifecycle".

> 
> The other question I have is if there's a way to make the debug view show everything, so if something _isn't_ being detected, I can see if it is there with a lower confidence or if it's just missing entirely. For example, if I'm seeing an object on one of my cameras that I want to be detected -- I don't know how to tell why it isn't being detected. I.e., do I need to train my Frigate+ model more/do I need to tweak the min score or threshold/is there something else wrong? I can keep adjusting the object parameters to try and get a detection, but again, that feels like a big guessing game and like I'm missing an obvious better approach.

No, this can't be done explicitly, but what you're doing by lowering scores and looking for detections is the recommended method.

> 
> I'm feeling very limited to needing to try and observe things live to view this sort of information, so I feel like I'm missing something obvious here and taking the wrong approach to tuning.

You're doing great. Frigate 0.15 will make this even easier.

**Reply by baffles** [reply]
Ah great, looking forward to 0.15! Thank you!
